Use standard version of libmpeg2 instead of embedded one.
Resolve a problem with RPI2 and GCC 7.

Signed-off-by: Laurent Merckx <laurent-merckx@skynet.be>

--- a/src/CMakeLists.txt	2019-04-01 20:08:16.953920498 +0200
+++ b/src/CMakeLists.txt	2019-04-02 21:58:24.204989201 +0200
@@ -36,15 +36,16 @@ include(GNUInstallDirs)
 include(GetGitRevisionDescription)
 include(InstallRequiredSystemLibraries)
 include(FindPkgConfig)
-include(ExternalProject)
-include(BuildLibMPEG2)
+#include(ExternalProject)
+#include(BuildLibMPEG2)
 
 use_cxx11( )
 
 PKG_SEARCH_MODULE(SDL2 REQUIRED sdl2)
 PKG_SEARCH_MODULE(SDL2_TTF REQUIRED SDL2_ttf)
 PKG_SEARCH_MODULE(SDL2_IMAGE REQUIRED SDL2_image)
-build_libmpeg2( )
+PKG_SEARCH_MODULE(MPEG2 REQUIRED libmpeg2)
+#build_libmpeg2( )
 
 message(STATUS "Target: ${CMAKE_SYSTEM_NAME} ${CMAKE_TARGET_ARCHITECTURES}")
 
@@ -72,6 +73,7 @@ message(STATUS "Version: ${VER_TAG}")
 include_directories(
     ${PROJECT_BINARY_DIR} # config.h
     ${SDL2_INCLUDE_DIRS}
+    ${MPEG2_INCLUDE_DIRS}
 )
 
 if( "${CMAKE_TARGET_ARCHITECTURES}" MATCHES "i386" )
@@ -97,17 +99,17 @@ add_subdirectory(timer)
 add_subdirectory(video)
 add_subdirectory(vldp)
 
-add_dependencies( vldp libmpeg2 )
-add_dependencies( ldp-out vldp )
-add_dependencies( game vldp )
-add_dependencies( sound vldp )
+#add_dependencies( vldp libmpeg2 )
+#add_dependencies( ldp-out vldp )
+#add_dependencies( game vldp )
+#add_dependencies( sound vldp )
 
 add_executable( hypseus hypseus.cpp globals.h )
 target_link_libraries( hypseus plog io timer sound video cpu game ${SDL2_LIBRARIES} ${SDL2_TTF_LIBRARIES} )
 
 set_source_files_properties( testvldp.cpp PROPERTIES COMPILE_FLAGS -DSHOW_FRAMES)
 add_executable( testvldp testvldp.cpp )
-target_link_libraries( testvldp video vldp ${SDL2_LIBRARIES} ${SDL2_TTF_LIBRARIES} ${SDL2IMAGE_LIBRARIES})
+target_link_libraries( testvldp video vldp ${SDL2_LIBRARIES} ${SDL2_TTF_LIBRARIES} ${SDL2IMAGE_LIBRARIES} )
 
 if(BUILDBOT)
     add_custom_command( TARGET hypseus
--- a/src/vldp/vldp_internal.cpp	2019-04-02 22:00:27.879821269 +0200
+++ b/src/vldp/vldp_internal.cpp	2019-04-02 22:00:35.959745458 +0200
@@ -37,8 +37,6 @@
 
 #include <inttypes.h>
 
-#include <mpeg2.h>
-
 // NOTICE : these variables should only be used by the private thread
 // !!!!!!!!!!!!
 
--- a/src/vldp/vldp.h	2019-04-02 21:59:53.680142780 +0200
+++ b/src/vldp/vldp.h	2019-04-02 22:00:15.471937796 +0200
@@ -34,6 +34,8 @@ extern "C" {
 // Ubuntu Linux complains with plain <SDL.h>
 #include <SDL.h> // only used for threading
 
+#include <mpeg2.h>
+
 struct yuv_buf {
     uint8_t *Y;     // Y channel
     uint8_t *U;     // U channel
